src/controllers/oauth_controller.ts

```diff
@@ -109,6 +109,10 @@ function redirect(response: express.Response, baseURL: URL, params: {[key: strin
       url.searchParams.set(key, params[key].toString());
     }
   }
+  if (url.hostname === 'gateway.localhost') {
```

Comment:
`gateway.local`?

Reply:
I'm using gateway.localhost as a placeholder for redirecting back to the gateway. I chose the .localhost TLD to be reminiscent of gateway.local but not step on the toes of anyone who wants to use gateway.local in a client definition.